hudi-client/src/main/java/org/apache/hudi/table/HoodieTable.java

```diff
@@ -635,9 +635,9 @@ public boolean requireSortedRecords() {
     return getBaseFileFormat() == HoodieFileFormat.HFILE;
   }
 
-  public HoodieTableMetadata metadata() {
+  public AbstractHoodieTableMetadata metadata() {
```

Comment:
I would prefer to still have an interface for `HoodieTableMetadata` and return that here. We can have the abstract class internal to the actual implementation?

Reply:
I maintained this interface in revision after rebasing and then built `AbstractHoodieTableMetadata` on top of this.